src/net_processing.cpp

```diff
@@ -4851,6 +4885,28 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                             pto->m_tx_relay->filterInventoryKnown.insert(txid);
                         }
                     }
+
+                    // Populating local reconciliation set.
+                    if (peer->m_recon_state && txs_to_reconcile.size() != 0) {
+                        assert(peer->m_recon_state->m_requestor || peer->m_recon_state->m_responder);
+                        int64_t recon_set_overflow = peer->m_recon_state->m_local_set.size() + txs_to_reconcile.size() - MAX_PEER_TX_ANNOUNCEMENTS;
+                        if (recon_set_overflow > 0) {
+                            LogPrint(BCLog::NET, "Reconciliation set for the peer=%d is at capacity, not adding new transactions. \n", pto->GetId());
+                            // Since we reconcile frequently, it either means:
+                            // (1) a peer for some reason does not request reconciliations from us for a long while, or
+                            // (2) really a lot of valid fee-paying transactions were dumped on us at once.
+                            // We don't care about a laggy peer (1) because we probably can't help them even if we flood transactions.
+                            // However, exploiting (2) should not prevent us from relaying certain transactions.
+                            // Since computing a sketch over too many elements is too expensive, we will just flood some transactions here.
```

Comment:
ce83d3fb

I think we should have this discussion about just throwing away transactions under some `SKETCH_MAX_SIZE`. Sort them by feerate, keep the ones above the mark, discard the others.

We're already doing the same assumptions if a a transaction is under mempool min fee, we won't relay it further. That can be the same with reconciliation, we don't allocate sketch resources for low-fee transactions. 

That said, I feel it should be part of a wider conversation about how we price better network resources while still flowing through the network what is "near confirmation".